libpod/runtime_ctr.go

```diff
@@ -305,11 +306,16 @@ func (r *Runtime) setupContainer(ctx context.Context, ctr *Container) (_ *Contai
 				return nil, errors.Wrapf(err, "error retrieving named volume %s for new container", vol.Name)
 			}
 		}
-
+		var volOptions []VolumeCreateOption
 		logrus.Debugf("Creating new volume %s for container", vol.Name)
-
 		// The volume does not exist, so we need to create it.
-		volOptions := []VolumeCreateOption{WithVolumeName(vol.Name), WithVolumeUID(ctr.RootUID()), WithVolumeGID(ctr.RootGID())}
+		// Get appropriate user details.
+		uid, gid, _, err := chrootuser.GetUser(ctr.state.Mountpoint, ctr.config.User)
+		if err != nil {
+			volOptions = []VolumeCreateOption{WithVolumeName(vol.Name), WithVolumeUID(int(uid)), WithVolumeGID(int(gid))}
+		} else {
+			volOptions = []VolumeCreateOption{WithVolumeName(vol.Name), WithVolumeUID(ctr.RootUID()), WithVolumeGID(ctr.RootGID())}
```

Comment:
Why the fallback here? Do we expect this to fail?

Reply:
It seems like this might be to handle cases where `ctr.config.User` is unset, but we should explicitly handle that case, not do a catch-all for errors.